pkg/cmd/server/origin/ensure.go

```diff
@@ -22,3 +22,11 @@ func (c *MasterConfig) ensureOpenShiftSharedResourcesNamespace(context genericap
 	}
 	return nil
 }
+
+// TODO: We should have a single place where we register all post start hook
+// and make this function unnecessary
+func ensureAddPostStartHook(genericAPIServer *genericapiserver.GenericAPIServer, name string, hook genericapiserver.PostStartHookFunc) {
```

Comment:
@deads2k upstream doesn't really need it as they run all their AddPostStartHook() calls into a function that already fatals if an error is returned. So it seem like upstreaming an OrDie version there would be useless for upstream.

That's why we have a TODO to create also a registration function in origin as well that properly deal with it at some point.